### PR TITLE
Update token check URL, using the current one as the fallback

### LINKED ISSUE #27036: Update token check URL
We moved our token check service to a static IP, with a new domain, so customers can allow-list it in tighter environments.

Some people may have white-listed the old one, so in order to prevent disruption, this should entail:
- First trying the token check with the new domain (`token-check.metabase.com`)
- If that fails, try the old one (`store.metabase.com`) as a backup

That's it.

### PR DESCRIPTION
**Pull request by noahmoss**
Resolves #27036

### CODE REVIEW

File: src/metabase/public_settings/premium_features.clj [hunk 1]
```diff
@@ -21,16 +21,21 @@
...
    (when config/is-dev?
      (some-> (env :metastore-dev-server-url)
              ;; remove trailing slashes
              (str/replace  #"/$" "")))
-   "https://store.metabase.com"))
+   "https://token-check.metabase.com"))
+
+(def store-url
+  "Fallback URL for token checks, also used for fetching cloud gateway IP addresses"
+  "https://store.metabase.com")
```
[the remark this answers is not in the document]
Comment (imrkd) [reply src/metabase/public_settings/premium_features.clj:37]: @noahmoss afaik we can't really use metabase in devmode in our cloud either way so I don't think it helps us test anything.

[the remark this answers is not in the document]
Comment (imrkd) [reply src/metabase/public_settings/premium_features.clj:37]: btw I think we're likely going to do more `store <--> metabase` integration in the future so maybe the docstring could be more general.

File: src/metabase/public_settings/premium_features.clj [hunk 2]
```diff
@@ -21,16 +21,21 @@
...
+   ;; only enable the changing the token check url during dev because we don't want people switching it out in production!
    (when config/is-dev?
      (some-> (env :metastore-dev-server-url)
              ;; remove trailing slashes
              (str/replace  #"/$" "")))
-   "https://store.metabase.com"))
+   "https://token-check.metabase.com"))
+
+(def store-url
+  "Store URL, used as a fallback for token checks"
```
Comment (camsaul) [inline src/metabase/public_settings/premium_features.clj:36]: This isn't only used for token checks, it looks like it's also used for `cloud-gateway-ips-url`. Probably worth mentioning that here

**Review by camsaul — APPROVED** [review]
LGTM

**Pull request merged**